src/components/CheckBox/CheckBox.component.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import {Form} from 'react-bootstrap';
+
+import {CheckBoxProps} from './CheckBox';
+import {StyledCheckbox} from './CheckBox.styles';
+
+/**
+ * Checkbox Component
+ *
+ * @example Correct usage with default props
+ *
+ * ```ts
+ *  <Checkbox label="Label"/>
+ * ```
+ */
+export const Checkbox = (props:CheckBoxProps) => {
+    const {className = '', id, label, error, ...restProps} = props;
+
+    return (
+        <StyledCheckbox
+            type='checkbox'
+            className={`mm-checkbox ${className}`}
+            id={id}
+        >
+            <Form.Check.Input
+                type='checkbox'
+                className='mm-checkbox_input'
+                isInvalid={error}
+                {...restProps}
+            />
+            <Form.Check.Label>{label}</Form.Check.Label>
+        </StyledCheckbox>
+    );
+};
```

Comment:
```suggestion
export const Checkbox = ({className = '', id, label, error, ...restProps}: CheckBoxProps) => (
        <StyledCheckbox
            type='checkbox'
            className={`mm-checkbox ${className}`}
            id={id}
        >
            <Form.Check.Input
                type='checkbox'
                className='mm-checkbox_input'
                isInvalid={error}
                {...restProps}
            />
            <Form.Check.Label>{label}</Form.Check.Label>
        </StyledCheckbox>
    );
```
see if this works otherwise current is fine

Reply:
It'll work, but I think we should not expose props or props' default value in the functional component's argument.
I have destructured props directly in the function's argument for sub-components.